### PR TITLE
add recent contributor

### PR DESCRIPTION
**Pull request by chuenlok**
1. add recent contributor
2. add back the `add source` button to metric value

### CODE REVIEW

File: mod/wikirate/set/self/recent_contributor.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+
+def item_names args={}
+  limit = Env::params[:limit] || 5
+  Card::Act.select(:actor_id).distinct.where("card_id is not NULL and actor_id <> #{Card["wagn bot"]} and actor_id <> #{Card["Anonymous"]}").order('id DESC').limit(limit).map do |act|
+    Card[act.actor_id].id
+  end
```
Comment (ethn) [inline mod/wikirate/set/self/recent_contributor.rb:13]: no comment, no helpful method names...

at this point we need cleaner code.  a massive query like that needs its own method.

Why is a method named "item_names" returning ids?

File: mod/01_rating/set/right/metric_value.rb [hunk 2]
```diff
@@ -41,10 +33,8 @@
   view :timeline_header do |args|
     wrap_with :div, :class=>'timeline-header' do
       case args[:column]
-      when :credit
-        _optional_render(:timeline_add_new_link, args, :show) || ''
       when :data
-        timeline_head('Year','year')+timeline_head('Value','value')
+        timeline_head('Year','year')+timeline_head('Value','value')+_optional_render(:timeline_add_new_link, args, :show) || ''
```
Comment (ethn) [inline mod/01_rating/set/right/metric_value.rb]: can we clean this code up, too?  line too long.

**Comment by ethn** [comment]
is this code covered in any test?

**Comment by chuenlok** [comment]
Test is updated for that

File: mod/01_rating/set/right/metric_value.rb [hunk 3]
```diff
@@ -41,10 +33,10 @@
   view :timeline_header do |args|
     wrap_with :div, :class=>'timeline-header' do
       case args[:column]
-      when :credit
-        _optional_render(:timeline_add_new_link, args, :show) || ''
       when :data
-        timeline_head('Year','year')+timeline_head('Value','value')
+        timeline_head('Year','year')
+          .concat(timeline_head('Value','value'))
+          .concat(_optional_render(:timeline_add_new_link, args, :show) || '')
```
Comment (ethn) [inline mod/01_rating/set/right/metric_value.rb:39]: much more legible, thank you!

File: mod/wikirate/set/self/recent_contributor.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
+
+
+# this is pretending as a pointer card
+# to get the recent contributor, query the ids and names from the act table to get recent editors
+def item_names args={}
+  # may need different number of editors
+  limit = Env::params[:limit] || 5
+  # this where clause to get recent editors(except wagn bot and annonymous) and the acted card should not be null
+  where_clause = "card_id is not NULL and actor_id <> #{Card::WagnBotID} and actor_id <> #{Card["Anonymous"].id}"
```
Comment (ethn) [inline mod/wikirate/set/self/recent_contributor.rb:9]: "Anonymous" is a cardname.  Suggest Card::AnonymousID

**Pull request merged**